### ISSUE TITLE
[FEATURE] Follow public local timelines from other instances

### ISSUE THREAD

**Issue by alsternerd** [tootsuite/mastodon#6942]
Just what the title says.

Maybe implemented like the "show a toot from another instance"-feature: Copying the full URL of the instance into the search bar opens the local timeline of this other instance.

Which makes following interests easier, since a lot of instances are specialised on some specific interests, but a person does'nt follow only one. So to mix things a lot more up.

**Comment by Cassolotl** [comment]
Do you mean like, to make the local timeline of another instance a pin-able column? (This sounds awesome.)

**Comment by alsternerd** [comment]
> Do you mean like, to make the local timeline of another instance a pin-able column? (This sounds awesome.)

Yes, that's exactly what I think of! Only shorter and to the point, thanks.

**Comment by Cassolotl** [comment]
*Browse Other Instances' Public Timeline #1053* was closed without being implemented a few hours ago. That discussion was super messy and started a long time ago, before we had public timelines on /about pages, which partially solved the OP's needs.

This new issue is more specific and I think it should stay open.

**Comment by deutrino** [comment]
Duplicate of #979?

**Comment by Cassolotl** [comment]
Certainly could be included in #979! ("Configurable federated timelines aside from "local" and "federated"")

>it would be more clever to allow users to configure additional timelines by specifying the servers they want to see toots from. The server would no longer need a whitelist (although they might have the possibility to configure a default curated whitelist timeline), but could show the whole federated timeline when the users wishes to do so.

**Comment by nightpool** [comment]
If we're going to continue this discussion here, which is fine, I'll repost my earlier comment, which I think is still releavent:

> There are third-party website that will allow you to do this, but my opinion is that if you want to view the instance timeline of another instance as part of your day-to-day browsing of the service, you should sign up for an account on it. Otherwise you have this weird culture where a small number of users are "performing" for a much larger audience, and there's no way for the users viewing that timeline to "join in", and post things to that timeline. this leads to very awkward possible community issues.

**Comment by antoniotorresm** [comment]
@nightpool I agree with you, but I don't think having people to have multiple accounts is enough user friendly. Maybe some kind of subscription system? There are a lot of thematic instances and it's natural to want to participate in various ones without having a bunch of accounts.

**Comment by Cassolotl** [comment]
> There are a lot of thematic instances and it's natural to want to participate in various ones without having a bunch of accounts.

Groups (#139) would probably resolve that issue - toots about a subject would end up in your timeline whether you follow the person posting or not.

**Comment by deutrino** [comment]
I think it would be cool to follow scholar.social but I'm not a scholar and don't want or need an account there.

Other people might want to follow the timeline from the recently created cars/racing instance.

Etc.

**Comment by moreystraus** [comment]
I made a small donation on bountysource for this.

**Comment by sprite-1** [comment]
Having this feature would make following along multiple instances way easier.

**Comment by kasinjsh** [comment]
> 
> 
> Having this feature would make following along multiple instances way easier.

Otherwise it is hard to stay for example up to date whit FOSS news and discussions living on My own instance.

**Comment by sparr** [comment]
This issue would make more sense without "follow" in the title. "Follow" already has a specific meaning on Mastodon. Following a whole instance would be... interesting, but is not at all what this discussion is about.

**Comment by rickisen** [comment]
Could this be implemented by simply making the instance a toot was made on a searchable field, similair to how hastags work in the search interface?

So if you want to see how the timeline in "linuxrocks.online" looks like you'd type "@linuxrocks.online" in the search, and the latest toots made there would show up.

**Comment by aranud87** [comment]
+1, it is same to Relay : 

![instance](https://user-images.githubusercontent.com/2754191/52856001-72bb3a80-3123-11e9-8ae1-108762daa00d.PNG)

Relay add many many instances i don't want, it would be easier to add instances directly i want to show on wall for all users on my local instance.

**Comment by Cassolotl** [comment]
@aranud87 You might be interested in *Instance-wide subscription #8082*!

**Comment by donmasto** [comment]
"There are third-party website that will allow you to do this, but my opinion is that if you want to view the instance timeline of another instance as part of your day-to-day browsing of the service, you should sign up for an account on it."

This shows why Mastodon could fail. We need a setup where we may follow all human beings and orhganizations on earth. Why would you limit who you can follow? Imagine all the email accounts you would need ... sounds like a comunistic, bureaucratic nightmare. This is why capitalism is better than socialism

**Comment by Cassolotl** [comment]
>Imagine all the email accounts you would need

It's okay, you can use the same email address on all instances because they're totally separate websites. :)

Not needing to sign up to an instance in order to see and interact with its local timeline seems good though anyway.

**Comment by CodingEagle02** [comment]
Uh, since no one has mentioned this. If this feature was implemented, would you be able to post to remote instances?

**Comment by sprite-1** [comment]
> Uh, since no one has mentioned this. If this feature was implemented, would you be able to post to remote instances?

I would hope not, that's going to be ripe for abuse

**Comment by trwnh** [comment]
>>Uh, since no one has mentioned this. If this feature was implemented, would you be able to post to remote instances?
>I would hope not, that's going to be ripe for abuse

You don't post "to" an instance. You post to a collection of actors, like your followers, or to Public. You would need a separate account/profile on that instance, if you wanted your post to show up in that "local" public timeline.

**Comment by sprite-1** [comment]
> You would need a separate account/profile on that instance

I believe that's what the other guy was asking. If it would be possible to post to have your pots show up in a remote instance's timeline which is why I said it would be ripe for abuse

**Comment by CodingEagle02** [comment]
> 
> 
> > You would need a separate account/profile on that instance
> 
> I believe that's what the other guy was asking. If it would be possible to post to have your pots show up in a remote instance's timeline which is why I said it would be ripe for abuse

Yeah, that's what I meant. Also, the way I think of that problem is admins having the power to block users or instances from being able to post on their timeline. Thoughts?

**Comment by sprite-1** [comment]
I see that being akin to someone being able to send an email on behalf of your own server without them needing to have an account with you. I see the use-case but I personally wouldn't want it

**Comment by CodingEagle02** [comment]
That's true. I don't know, though. I see the whole thing - plus decentralysed identities - as a way to abolish the idea of instances as different websites and more as community spaces within the larger Mastodon platform. Similar to spaces and communities in other social media. For example, you don't have to create new accounts to see new subreddits or Facebook groups or to post to them. You just create one account and you can go wherever you want. I'd like to see that transition for Mastodon and hopefully the Fediverse as a whole eventually. After all, what's the point of a decentralysed platform if you're bound by whatever community you chose?

**Comment by CodingEagle02** [comment]
Hypothetically, instances could always disable such a feature. So for example, someone running an instance on their house wouldn't need to worry about people abusing their server, but at the same time, they would not be bound by it and could join multiple communities in the comfort of their customisable private space.

**Comment by sprite-1** [comment]
So you're looking for a more outward than an inward approach to things. Also I'd like to add more to my "abuse" perspective. A group of people can potentially harass both instances by making a bunch of account on one instance and spamming the other, one of the common knee-jerk reactions would be to block the whole instance that hosts multiple spam accounts which will be a disservice to the other users of said instance as instead of only not being able to participate on the remote instance's timeline, now they're cut off completely. Not to mention each instance has their own policies of what they are comfortable and not comfortable with allowing people to post (nudity, gore, etc). Rules which are only really properly shown when signing up.

The way I see it is that, currently, Mastodon takes the more reserved approach wherein you can follow anyone to tailor your content for your personal timeline but you can't affect others'.

**Comment by marrus-sh** [comment]
**RE: “Participating on instances where you do not have an account”:** There are people working on this problem but those people are not the Mastodon development team and those technologies are probably a few years out.

So, uh, be patient on that one.

**RE: “Pinnable remote-local timelines”:** The major thoughts I have are:

1. Instances do not receive every post from every timeline by default. There is presently no good way for an instance to just “subscribe to all the public posts of another instance”. With the new instance Actors (#11321), this could hypothetically be achieved (having one instance Actor subscribe to another), but #8082 is definitely a prerequisite if you don't want these timelines to just be “public posts that happen to have already appeared on our Federated Timeline for other reasons”.

    One of the reasons why “viewing the public timeline page visible on the instanceʼs URL” is currently required is because it is the most *complete* view of the public timeline that exists. Anything federated loses some information. This is of course less convenient, but having deterrents for this sort of thing might actually be a good thing, because…

2. I agree with @nightpool that this might lead to weird community behaviours, specifically from the direction of “context collapse”. For people on small (especially single-user) instances, there is a pretty good idea of who will see their public posts (their followers, and the other people on the instance's followers). This expectation gets broken when posts get boosted / shared around a bunch, but it's *good enough* that like, I can probably post something publicly on `gentle.town` (where I live) and, unless it goes viral, not get weird randos that are completely disparate from my social circles viewing it. If people can subscribe to gentle.town (à la #8082) and view its posts, then the local understanding of who might see them gets a *lot* more confusing, especially since (as far as has been discussed) there is no public announcement that Stranger $X is now viewing everything your instance has to say.

    This can be mitigated on instances which implement local-only posting (which explicitly defines a limited context for the posts), but as this is not an upstream feature, it is not relevant to the discussion here. I would say that implementing this *without* local-only posting effectively threatens the public timelines with becoming an entirely contextless space, to the detriment of the local communities.

**Comment by trwnh** [comment]
>abolish the idea of instances as different websites [...] you don't have to create new accounts to see new subreddits or Facebook groups

this is not happening any time soon -- you are posting web resources to a website, not to a group or to a subreddit. people follow other people, not topics, and local timelines don't strictly have a topic. it's just an arbitrary way to filter a subset of public posts. if you want addressable groups, then that's #139

>what's the point of a decentralysed platform if you're bound by whatever community you chose?

by signing up on a specific website, you are choosing a service provider. the admin of that service can run that service as they wish.

**Comment by fsnk** [comment]
> > > Uh, since no one has mentioned this. If this feature was implemented, would you be able to post to remote instances?
> 
> > I would hope not, that's going to be ripe for abuse
> 
> You don't post "to" an instance. You post to a collection of actors, like your followers, or to Public. You would need a separate account/profile on that instance, if you wanted your post to show up in that "local" public timeline.

Obviously posts from a remote server wouldn't show up in a local timeline, that would defeat the entire purpose. I assume this would work like any other timeline, and you'd be able to reply to a post and people who expanded the conversation would see your replies in that context. 

For posts to appear top level in the local timeline on a server, they should posted by a local account on that server.

**Comment by bs2kbs2k** [comment]
I'm not a expert, but can't it be implemented fully client-side by fetching the other instance's /public feed?

**Comment by BakedCrossiant** [comment]
Any update on this?

**Comment by ykgoon** [comment]
I upvote this.

**Comment by eobet** [comment]
Crossposting here too...

I would love to be able to add the local timeline of a different server to a column in the advanced web layout!

**Comment by FediVideos** [comment]
I think adding Local feed following from other servers would be really good for the Fediverse because current system is causing confusion and stress for new users.

Feedback from new users in current and previous wave is very commonly this:

They see servers as subreddits, and want to be able to "browse other servers". So, they think they need to create lots of accounts, and they find this "need" for multiple accounts as confusing and stressful.

I've tried explaining that they don't need multiple accounts, that content will come to them by following people from other servers, but they still think of other servers as something they're missing out on if they aren't a member. There's a lot of FOMO, I've had this same conversation again and again with new users.

If users could follow other servers' local feeds, they would no longer think they need multiple accounts. This would make the Fediverse seem much simpler, and it would also make smaller servers more attractive which would make the Fediverse healthier.

**Comment by coffeehabit** [comment]
Creating a list of other servers to browse a curation of instances would be pretty cool. "Tech servers" "Science servers" etc

**Comment by dobkeratops** [comment]
could a similar effect be created with a default agregator bot that the server software includes (which server admins can disable or replace as an optin) ... and servers could post a list (visible in user UI) of recommended follows .. easy to click/browse see, included in the channels feed by default (local or federated or both?) -  which the user can mute; (then is down to admins to balance: how many of these to show).

would this be any less UI/software work than adding user server follows?

I also really want something like this. 
The underlying federated idea is awesome! but it does seem to take an extra layer of UI that the world hasn't figured out yet

**Comment by dobkeratops** [comment]
Currently to build a varied feed you need to follow randomly:
problematic for 3 reasons:

[1] according to social media norms this just feels *weird*, almost inappropriate - you're expressing a signal more specific but it's meaningless. (hence user discomfort)

[2] it's making the information encoded in the mastodon ecosystem less valuable, because it's corrupted with this noise. you're no longer getting an *accurate* graph of who genuinely wants to follow who.

[3] there's usually extra clicks involved in finding and following profiles outside of your feed (if thats avoidable , then there's extra UI needed to fix).

This is why the system desperately needs a different way to broaden the feeds.

I dont want to sound critical of course. it's a voluntary project if I've understood right , the underlying ideas are great, and are worth supporting. It's more like "this is what this system needs to do itself justice.."
I find myself in a similar situation with the Rust programming language. It's my favourite, I've put *a lot* of effort into switching from C++, but many established C++ users just get put off by the early experience (<1000 lines it feels cumbersome) , as such they never get to appreciate it's virtues ( it really shines as a program grows)

**Comment by eobet** [comment]
Since there's a _significant_ migration currently happening, I'm observing a lot of feedback from those who don't that they don't want to have to miss out on the timelines of other servers.

Now, like it or not, there are a lot of Mastodon instances which advertise themselves as having some sort of specialization, and people with a lot of followers on the bird site often only specialize in one thing. But most people aren't like that. Most people have more than one interest, and thus want to follow what's going in some place other than the server they happened to create an account on (the other huge complaint I see is that people have actually registered an account on each server they're interested in and are having a difficult time managing that, but that's outside the scope of this topic.

Also, the firehose of the Federated timeline might be feasible to drink from now, but with 10x or 100x more growth, that's going to be just noise (if it isn't already).

Exhibit [A](https://twitter.com/ashtonpittman/status/1588626168391127040?s=20&t=0Vv80PUNL4fYxOyBT9yIcQ) [B](https://twitter.com/mdf200/status/1589304365814546434?s=20&t=QkI3AhZxOqeIzXudCu_KAw) [C](https://twitter.com/UrbanDemog/status/1588641647805374464) [D](https://twitter.com/erinbiba/status/1589054628788047872?s=20&t=QkI3AhZxOqeIzXudCu_KAw) [E](https://twitter.com/_nyancrimew/status/1589264764450275328?s=20&t=QkI3AhZxOqeIzXudCu_KAw) [F](https://twitter.com/archasa/status/1589356614145912832?s=20&t=cmk4uZ1sVsIYqPekqXTGWw) [G](https://twitter.com/nickf/status/1588566993325953024?s=20&t=utuz0D-px7x05IIT0JMTUQ) [H](https://twitter.com/VeryBadLlama/status/1589086085380190209?s=20&t=QkI3AhZxOqeIzXudCu_KAw) [I](https://twitter.com/glassbottommeg/status/1588600614053117953?s=20&t=QkI3AhZxOqeIzXudCu_KAw) [J](https://twitter.com/aurahack/status/1588608019839209472?s=20&t=QkI3AhZxOqeIzXudCu_KAw) [K](https://twitter.com/thesiswhisperer/status/1588647643244560384?s=20&t=QkI3AhZxOqeIzXudCu_KAw)

I think there's an opportunity here to turn the specialized instances into something positive for everyone...

**Comment by brendanjones** [comment]
[This one](https://twitter.com/glassbottommeg/status/1588600614053117953) from @eobet hits the point home for me
> Something folks moving to mastodon may not entirely realize is that server choice can make your experience better. You probably never use Federated timeline, but Local timeline? That's your local server. Which, if it's cool, can be useful!

Federated timeline is interesting (and often amusing) for an occasional visit, but honestly it's a firehose of posts that mostly aren't of interest to me. That's only going to become more true as more people join the fediverse.

How I actually use timelines:
1. Hear what the specific people I follow are saying: home feed, or lists
2. Discover what everyone is saying about specific topics: follow hashtags
3. Discover what people in my wider network are talking about, not specific to any topic (a broader discovery): local timeline

(3) is currently limited to the one server you joined, if I understand correctly. That's frustrating because I know there are a couple of other servers out there in which I'm interested and I'd love to browse for new content, and find people to follow. My example: I'm on fosstodon.org for my IT interests, but I'd also like to follow climatejustice.social. 

One big caveat to all that: the goal here (for me) is to easily browse other servers, and follow people on those servers. Browsing other servers is already easy, especially with v4.0. You load up an instance example.instance and it'll redirect to example.instance/explore. Great!

However you can't follow people easily. One of the big blockers when viewing another server's feed is the number of steps to follow someone. It's literally 6 steps -  it should be two clicks (click on name, click 'follow'). Solve this, and you've solved 90% of the barriers here. 

Yes I'd love to save other servers as timelines (lists?) in my own server UI so I never have to leave that one UI, but that's a nice to have.

**Comment by philpax** [comment]
Just weighing in on this as someone who just migrated today: the Local view of my instance is fantastic because it lets me discover what people are up to on my "specialised" instance, but I *really* want to see what people are up to on other specialised instances I'm on!

The Local view has already helped me discover new people and grow out my Mastodon network. I'd like to be able to easily find new people and toots from elsewhere, too.

**Comment by mitar** [comment]
Maybe an alternative could be that you could have multiple accounts on multiple instances (so that you can also participate in those instances, not just lurk, and also observe potentially private local feeds), but that federation would know that this is all the same account globally and allow anyone to tag/follow under any handle, getting a sum of all your accounts back. So the downside of multiple accounts is that followers have a hard time know which handle you use. So there should be some way to make it not matter whichever handle they pick, it would all be the same global account.

**Comment by brendanjones** [comment]
> So there should be some way to make it not matter whichever handle they pick, it would all be the same global account.

@mitar oh gods, don't open that can of worms haha. I was thinking about this a couple of days ago, that what you actually want on Mastodon (and even the Fediverse) is a universal user ID for each person. Doesn't matter what server you're on, you maintain the same ID. It'd solve so many issues. I have a feeling I'm not the first person to have this idea, but I'm really new here so no idea the history behind why this wasn't done. Anyone got links to discussions? I didn't find anything so far.

**Comment by mitar** [comment]
OK, having unique handle over multiple servers is tricky and a can of worms. But having metadata on your handle saying "BTW, those other handles belong to me" (with some proof) should be easier. And then only client combines those together, while at the API level things could stay probably the same.

**Comment by FTWynn** [comment]
It seems largely already in place with the redirect flow, right? Just don't terminate so much on the "remote" end when pointed back to the "home" instance.

Or maybe we could have the ability to ACL any other signed in account, even on a different server, to an account. 

Re-aggregating the activity then becomes more of a challenge, but it would be easier to explore different instances.

**Comment by Rhyothemis** [comment]
How about having administrators select server categories (topical, geographical) and then have a feature where the federated timeline could be filtered by server category. So for example one could filter the federated timeline by 'science' or 'North America'? Is that feasible?

**Comment by ipatrol** [comment]
I'd like to toss my hat into this ring. Some instances have strict rules about who can sign up for them in order to stay topical, but then it makes it difficult for those interested in that topic to follow along.

**Comment by gcrow** [comment]
Local timelines are viewable via public GET calls by default - just needs a client - https://docs.joinmastodon.org/methods/timelines/

**Comment by StyXman** [comment]
Coming from RSS readers, I would think an instance's local timeline (LTL) an aggregate of the instance's users, like a RSS planet. So, I could follow my instance's RSS and remote instances RSS in a few clicks instead of following each user one by one. Granted, this would not allow to later 'defollow' individual users.

Or maybe yes: don't actually follow the instance's (local or remote) TL but just follow all its users, but in one go, not one per user.

**Comment by NurseTonyF** [comment]
I would also like this feature as described. 
Currently:
•I see peoples' posts from other interesting instances. 
•I visit that instance by typing it in a browser
•if I want to interact (comment, boost, follow the person), it asks me to sign in to that instance or to copy and paste the specific link to that person/post in my own logged in instance
•if i type the broad instance into my search box in my own logged in instance, it only brings up a list of people or posts on that instance, not the timeline of that instance
Desired:
•log in to own instance
•search broad instance timeline by typing in the instance 
•mark the instance somewhere for future perusal instead of manual typing
my current workaround is quite clunky
•create a mastodon bookmark folder on chrome to bookmark instances' timeline pages
•have to copy and paste back in my own logged in page to be able to interact

Either of two solutions would improve the situation:
•enable "bookmarking" of timelines on our own logged in pages
OR
•improve/create ability to log in with our own instance credentials when perusing other instances from their xx.yy timeline site

**Comment by JasonKeirstead** [comment]
As a user, I would like to be able to configure my "Local" feed to include more than one server. Being able to "follow" an entire other server's feed, could also fill the need. The core need is wanting to see public posts from multiple servers in one place without having to jump all over the place.

As a mastodon admin, I would like to be able to "Peer" my server with others in a similar interest domain, so that *all users of my server* have the posts in those other servers show up in their feed without having to do anything at all.  This could show up under a new feed area called "Peers" or something. It could also simply augment the "Local" feed.

**Comment by cobordism** [comment]
@NurseTonyF It's because pettingzoo.co does not expose its local timeline.
see, both https://pettingzoo.co/public and https://pettingzoo.co/public/local are empty.

It has nothing to do with your own instance's relationship with the remote instance. It is only about what the remote instance exposes.

**Comment by kekoziar** [comment]
> Since there's a _significant_ migration currently happening, I'm observing a lot of feedback from those who don't that they don't want to have to miss out on the timelines of other servers.
>  [...]
> 
> Also, the firehose of the Federated timeline might be feasible to drink from now, but with 10x or 100x more growth, that's going to be just noise (if it isn't already).
> 
> Exhibit [A](https://twitter.com/ashtonpittman/status/1588626168391127040?s=20&t=0Vv80PUNL4fYxOyBT9yIcQ) [B](https://twitter.com/mdf200/status/1589304365814546434?s=20&t=QkI3AhZxOqeIzXudCu_KAw) [C](https://twitter.com/UrbanDemog/status/1588641647805374464) [D](https://twitter.com/erinbiba/status/1589054628788047872?s=20&t=QkI3AhZxOqeIzXudCu_KAw) [E](https://twitter.com/_nyancrimew/status/1589264764450275328?s=20&t=QkI3AhZxOqeIzXudCu_KAw) [F](https://twitter.com/archasa/status/1589356614145912832?s=20&t=cmk4uZ1sVsIYqPekqXTGWw) [G](https://twitter.com/nickf/status/1588566993325953024?s=20&t=utuz0D-px7x05IIT0JMTUQ) [H](https://twitter.com/VeryBadLlama/status/1589086085380190209?s=20&t=QkI3AhZxOqeIzXudCu_KAw) [I](https://twitter.com/glassbottommeg/status/1588600614053117953?s=20&t=QkI3AhZxOqeIzXudCu_KAw) [J](https://twitter.com/aurahack/status/1588608019839209472?s=20&t=QkI3AhZxOqeIzXudCu_KAw) [K](https://twitter.com/thesiswhisperer/status/1588647643244560384?s=20&t=QkI3AhZxOqeIzXudCu_KAw)
> 
> I think there's an opportunity here to turn the specialized instances into something positive for everyone...

Being able to filter the federated timeline by server would be keen.

**Comment by JasonKeirstead** [comment]
@cobordism I tried using Fedilab, but the app was so glitchy I could not get properly authenticated. IMO this is still less than ideal because everything is in tabs and not sorted by time. I want *one feed* with 10 servers. That is not the same as 10 tabs that I need to constantly click through.

**Comment by cobordism** [comment]
> @cobordism I tried using Fedilab, but the app was so glitchy I could not get properly authenticated. IMO this is still less than ideal because everything is in tabs and not sorted by time. I want _one feed_ with 10 servers. That is not the same as 10 tabs that I need to constantly click through.

Ah. Then you and I have different use cases.  I want the separate tabs, because the instances I follow have very different specific topics/communities.

**Comment by binarydigitz01** [comment]
Right now I feel like there isn't really a good way to interact with another instance. With the new feature of following hashtags, you could possibly follow the hashtag that the instance most uses. You could also use it so that people of the other instance could see your post. 

Another way to solve this is to have 'remote accounts'. Basically you would have another account on that instance, but you would tell the other instance that this is a special account. It would be authenticated with your main account. And the remote account's profile would tell others whose remote account it is. You could choose to follow a specific remote account, if you're only interested in the particular topic of that person, or choose to follow the main account, which would notify you about tweets from the remote account as well. 
Only main accounts would be able to make a remote account.

**Comment by JasonKeirstead** [comment]
> Ah. Then you and I have different use cases. I want the separate tabs, because the instances I follow have very different specific topics/communities.

The same is true of me, but this is also how most social networks work... you have one feed, personalized by you, and it has all of the things you care about in it in a timeline... not 15 different feeds to monitor.

**Comment by neffscape** [comment]
Today I gave up and I created another account just to be able to find people to connect with in another instance. It felt wrong, like being two different persons in two separate lives. I really thing this is a much needed feature. Following other local timelines should be as easy as searching and following an hashtag. Please consider adding this feature in a future update. Is there a way to support or sponsor this particular development?

**Comment by Terra100f** [comment]
I hope this is solved. Lots of interesting solutions above. It would be helpful overall for society to be able to view each other more transparently instead of siloing ourselves in without options to look out. I vote for:

1 - choose your instance, set up your profile
2 - be able to view and sort all other instances and follow people from your instance without other steps
3 - be able to search with additional terms (any words) in the search box (not just hashtags, user name, URL (post/user)) -- I realize this suggestion is more Twitter like, but it's necessary for finding topics and where the conversations are happening
4 - allow pinning another instance's feed to a column AND/OR as a bookmark -- prefer as a column, so that if there is an instance that I want to follow regularly then I can keep track of the posts in the feed better with a column

I do not like having to create new accounts to be able to view other instances. This is madness.

**Comment by binarydigitz01** [comment]
I just had another idea, maybe implement a hashtag like system but for instances? lets call it an instance-hashtag for now.
Each instance would have it's own instance-hashtag, and you could follow and do regular hashtag things with it. all toots could only have 1 instance hashtag, and if toots use it, that toot would not appear in the local timeline of the instance in which the account was created. Instead it would appear in the local timeline of instance who owns that hashtag.

**Comment by brendanjones** [comment]
Hey so I'm seeing some nuance to what different people want to achieve in comments above. The issue name is focussed on one solution but I'd like to step back for a moment and take a higher-level look at user needs here. I've added a number of different needs and potential solutions below, and I'm curious which resonate with people. The solutions don't need to be/shouldn't all be implemented, because they're overlapping. It really depends on which user needs we do/don't want to solve for.

- I always want to view multiple instances at once
- - Allow adding multiple instances to local feed. No filtering required.
- I sometimes want to view other instances together with my local feed
-- Allow selecting multiple instances for local feed, and have ability to (de)select which ones you want to view at any one time. 
-- 
- I sometimes like to view other instances, not necessarily combined with my home feed. It's mostly the same instances each time.
- - Could be solved with local feed solutions above
- - Could also be solved by adding instances to lists, or 
-- Saving them as a separate list (and pinning a feed in advanced view is also possible, but that doesn't solve for all users)
- I sometimes like to view other instances, as exploration. It's not always the same instances.
- - Could be solved by some of the other solutions above, but also could be solved by opening an instance feed from search. It'd be the single feed you're looking at in simple view, and it'd add a column in advanced view. You would not be able to save/pin the view.
- - Could also be solved by adding a search and/or filtering of instances to federated timeline.

Hope this is helpful.

**Comment by Terra100f** [comment]
BrendanJones -- lots of great possibilities. 

For me, I would like to see the EXPLORE option greatly expanded as a view across 1) my instance, 2) instances I choose, 3) all instances.

However, to make that usable, for me it would have to offer trends. Because a feed of instances can be daunting to parse or make use of - thousands of posts a minute. So I feel it needs to be combined or offer the option of viewing trends, keywords, hashtags in those instances. 

I also would find it very useful to have not only hashtags but also **keywords** to be able to be pinned to a column AND to list and have it cut across all instances. What I'm trying to achieve is similar to the explore option in Twitter where Twitter had categories of topics and trending conversations in those categories. I could choose to add those categories to the "For You" tab. Granted, those were algorithmic, but it gave me a few minutes of interesting things to peruse. 

With everything from above, and I think many here would agree with me, there is the TIME EFFICIENCY factor. If you have the time to sit for an hour and go through a whole feed of something, then fine. But otherwise, I'm often looking for something specific, interesting conversations that are trending, being part of a group for a social movement or interesting question, or a quick discovery of something new. I don't have five hours in the day to do all this. So as much as we can talk about hating on the algorithms, having the option for them (which is most of the time) is a necessity because we are all strapped for time. And being able to have the options of being siloed in a group OR going to the whole set of instances would solve, I think, a lot of the varied needs here on Mastodon.

**Comment by brendanjones** [comment]
Actually to build on what you're saying there, I'm realising I focussed only on instances. There are multiple other ways of discovering and/or filtering content, especially hashtags and following specific people.

I feel like these changes we're discussing need some vision - what's the plan? Because it's technically possible to add searching and filtering for instances, hashtags and people to every single timeline and list. But is that a good idea? Probably not. Especially not if the features are different depending on what timeline you're using. Having inconsistent features across timelines is the kind of unexpected UX that doesn't match the "don't make me think" mantra.

So alternatively you could say to keep the default views the same always (Home is followed people, Local is your instance, Federated is federated), and if you want to create custom views then that's what lists are for. Add instance, hashtag and person following and filtering to lists, but don't add it to the default views. This is conceptually simpler - if you want a custom view there's only one place to go.

You could also do a hybrid version of that: make your home timeline customisable, and lists customisable, but keep local and federated views unchanged.

tl;dr before adding functionality to follow instances to your home or local timeline, it'd be good to decide the vision for customising timelines. Anyone got thoughts on that?

**Comment by Terra100f** [comment]
This is the vision I have in my head - file attached
[mastodon vision.pdf](https://github.com/mastodon/mastodon/files/10073077/mastodon.vision.pdf)

**Comment by brendanjones** [comment]
@Terra100f  love to see someone laying out their vision ❤️ A question: it looks like you’re adding a menu item there, “communities and topics”. Is there a reason why you wouldn’t bundle that functionality into lists? To me lists are the existing place where custom content belongs, and all the functionality you listed there could live together under lists, instead of having two locations for custom content.

**Comment by Terra100f** [comment]
@brendanjones The vision was sketched based on UX only. My brain puts communities and topics into an engagement feeling. Lists, well, are lists -- feel more like just feeds, not communities. 

Under a technical viewpoint, yes, you could lump the communities/topics into lists, but that's not UX. Which is why I imagine Twitter was created with both -- lists separate from communities/categories/topics. Of course, Twitter also utilized communities/categories/topics for algorithm driving. However, that's not necessarily bad in that space because isn't that how we function in real life? Think about a dinner table conversation. There is the ebb and flow of topics that bubble up and draw attention, sometimes for long periods. Essentially, a centralizing of thought which has exceptional connection moments.

If we go back to the original topic for this feature request "follow public local timelines from other instances" -- I'm digging deeper into why there would be such a request. Simply following timelines from other instances could bring some insight, but unlikely to bring connection or a feeling of community without a fuller set of features and engagement possibilities. And that's why I drew up the vision sketch to integrate this feature request into actual UX for public benefit.

**Comment by Terra100f** [comment]
@sparr Yes, I agree that reducing the requirement to follow would be an improvement in Mastodon and would relate to this overall problem of finding a balance between de-centralized vs. centralized. 

In Twitter, I appreciated not having to follow everyone I wanted to add to a list or view topically/community-wise.

**Comment by cobordism** [comment]
I have been discussing this with people on Mastodon too.
The tension comes about because of the dual role that instances play. On the one hand they host the user's account and provide access to the rest of the network, but on the other they also represent communities (via the local feed).

This issue "follow local TL from other instance" tells me the user wants to be part of the community (at least partially - being able to read and reply to topics) without having to move their account. In particular, we want to be able to read and respond to posts in many communities while using just one account. When I bookmark the local feed of another instance as a tab in my app, it does feel like a "group" might in other social networks. 

Aside: I can implement this myself on the client side by automatically following every account on the remote instance and maintaining a list of all those accounts (and also maintaining a list of all other accounts I follow so that this can act like my home feed would have before).. etc etc.  It's all possible but it is very clunky. 

A more fundamental question is whose responsibility it should be to retrieve the data. Should my instance get the posts from the remote instance and show them to me (just like it does when I follow someone), or should I directly get the posts from the remote instance (like when I go to instance.tld/main/local in my browser). 
I recon there is no reason to involve your local instance in this at all. .. at least not until you boost, favourite, or reply to a post.



Disclaimer: I never used topics on twitter and I hated the 'trending' system. Thus I really cannot comment on all that. 
I have just found that there are many interesting communities on Mastodon and would like an easier way to keep up with and take part in the conversations happening therein. Also, when I see someone make interesting posts and their instance seems new to me, I like exploring their instance to see whom else I might want to follow.

**Comment by RedSnt** [comment]
I've been wondering about this feature as well.  A simple whitelist or blacklist sorting against the federated timeline would do wonders, but might not be the most simple way for a regular user to interact with the feature. A simpler way of filtering would be to assign hashtag-like identifiers and be able to search for like "#Viv+#NJB+#GMD" which would gather timelines from Vivaldi Social, NotJustBikes.com and Gamedev.place as an example. I really like my pins and lists, so something like this would be incredible.

**Comment by orenTal** [comment]
@nightpool , the problem is that being part of a community and following a community is not the same thing. 
I mean I can not join journa.host because I am not a journalist. Also, I don't want to. But I may want to follow this server from my server.

Your point is based on the idea that following a community and being part of the community are the same thing. They are not.

**Comment by brendanjones** [comment]
There are multiple solutions that would achieve the same aim of exploring/following the timeline of another instance that this issue asks for. I've listed them all below. Where someone has already created an issue for that specific solution I've linked the issue. Let me know if I've missed any.

### Why did I do this? Because 

1. We need to decide which should or shouldn't be done.
2. It'll help talk about specific solutions, because just about every single one of these has been proposed above. Now you can reference a solution number, yay!

### Possible solutions

1. Filter the federated timeline on instance. https://github.com/mastodon/mastodon/issues/20995
3. Add instances to your local timeline (Instance filtering would also need to be added so you could customise which you see). https://github.com/mastodon/mastodon/issues/8224
4. Add instances to your home timeline (Instance filtering would also need to be added so you could customise which you see). https://github.com/mastodon/mastodon/issues/20995
5. Add instances to lists.
6. Search for an instance and explore it. Would display as the currently viewed feed in simple view, and a new column in advanced view. This is an incomplete solution as long as pinning only works for advanced view. Mockup of it [here](https://github.com/mastodon/mastodon/issues/1053#issuecomment-299669604) in https://github.com/mastodon/mastodon/issues/1053. The advanced-view-only version of it is here https://github.com/mastodon/mastodon/issues/19307
7. Add a new timeline specifically for following instances other than your own. AKA a 'home' feed for the instances you follow. https://github.com/mastodon/mastodon/issues/20379
8. Whatever this is actually asking for https://github.com/mastodon/mastodon/issues/979

Cast your votes ;)

**Comment by brendanjones** [comment]
My personal votes

1. Useful for instance exploration so I'd be happy to see this implemented anyway, but federated timeline is not where I'd want to permanently follow specific instances. 
2. Neutral. I think better to keep it simple and keep local as only ever your local instance.
3. I could be convinced of this one but I think it better to keep it simple and rather use Lists for custom feeds.
4. **My preferred solution - if lists were fixed in advanced view so that they opened a new column instead of covering the menu. This solutions offers maximum flexibility because lists are the place for all custom feeds.**
5. Being able to search for an instance should be added anyway. Also saving that instance is useful, but where do you save it in simple view? Personally I think pinned columns in advanced view are unnecessary duplication of functionality, and their functions should be rolled into Lists.
6. Not necessary, better done with Lists.
7. No idea.

> My brain puts communities and topics into an engagement feeling. Lists, well, are lists -- feel more like just feeds, not communities. Under a technical viewpoint, yes, you could lump the communities/topics into lists, but that's not UX. Which is why I imagine Twitter was created with both -- lists separate from communities/categories/topics.

@Terra100f iiuc you feel that lists don’t feel like the place to go for those topics. But when you think about viewing a community or topic, what would you expect to see? A feed of it, right? So to me the issue isn’t that communities and topics aren’t feeds, but rather that you don’t expect to explore those kind of feeds from Lists. And that makes it a naming/education issue. What if lists were changed to be called, say,  “Custom feeds”? Would that be the place you’d expect to be able to create and save custom feeds, which could include specific topics? And, to bring it back to this issue, ‘Custom feeds’ would also be where you’d follow other instances?

**Comment by sparr** [comment]
@brendanjones Your list omits the critical distinction of whether the data is coming from your instance or from the remote instance. In the former case, you would only see content that has been federated to your instance. In the latter case you would see all content, but sacrifice privacy. As mentioned in my comments and linked issue above, this is a critical decision that impacts almost every variation you've suggested.

**Comment by orenTal** [comment]
@sparr why is data coming from remote instances create problems for privacy?

**Comment by cobordism** [comment]
Let me try to tease out which ones require changes to the server, and which ones can in theory be done entirely on the client side.

> 1. Filter the federated timeline on instance. 

Could be done in the client side - i.e. in the app or frontend being used. It is not strictly necessary for the server to offer a filtered feed for this to work.

> 2. Add instances to your local timeline (Instance filtering would also need to be added so you could customise which you see).  

Perhaps not "the" local timeline, but an app/client could show an alternate home timeline with posts retrieved from multiple instances. So this too is possible without changes to the server. (however - see Note at the bottom of point 5)

> 3.  Add instances to your home timeline (Instance filtering would also need to be added so you could customise which you see).

Could be achieved by mass following combined with filtering, or as in 2. the app could simply synthesise a new 'home feed' with the desired posts. Might be impractical though as it would mess up the home feed when using other interfaces

> 4.  Add instances to lists.
I know nothing about lists, and have no idea what this means.

> 5. Search for an instance and explore it. Would display as the currently viewed feed in simple view, and a new column in advanced view. This is an incomplete solution as long as pinning only works for advanced view. 

This is two different things.

 "Search for an instance and explore it" is a potentially very useful feature. Currently the only implementation seems to be using your browser and opening the remote instance directly. 

"Would display as the currently viewed feed in simple view, and a new column in advanced view.  This is an incomplete solution as long as pinning only works for advanced view. " - this is a very specific implementation in one particular client/web interface.

Note:
The question is, do we want our own instance to retrieve all the posts from the remote instance and then offer them in a feed to their users, or do we just want the app/client/frontend to retrieve the posts directly from the remote instance, bypassing their own instance entirely?

Some instances do not expose their local timelines for browsing. If an instance does make the timeline available, any client/app can pull data from there and display it to the user. However, for instances that do not make their local timeline available on the web, my client would need to query my own instance to see which posts from the remote instance it knows about. (I suppose this is then equivalent to what I would see when filtering the federated timeline).

> 6. Add a new timeline specifically for following instances other than your own. AKA a 'home' feed for the instances you follow.

This is what I use now in the fedilab app. It works only for those instances that share their local feed via the web. 


------

Observations: 
a. Almost everything here can be done on the client / web interface side of things, although some might be better handled by the server. 
For example: maybe it is better for my client to request from my instance all posts from the federated timeline for which the poster's instance is X instead of requesting from my instance all posts from the federated timeline, and then simply filtering out those I don't want to see. 
The former requires changes to the server and api, the latter might waste resources / bandwidth.

b. should the client pull data directly from remote instances, bypassing the home instance?
With the browser I can go to https://chaos.social/public and https://journa.host/public/local and so any client should be able to display the same data. 
However, in this particular example, if I am a user of one of those two instances and I am only talking to my own instance, I will not be able to retrieve any posts from the other because chaos.social defederated journa.host this week. 
If the two instances in question are talking to each other, we have a choice. We can either pull the data from the remote instance or we can ask our own instance to pull the remote data and show it to us. 
I personally prefer the former because it is closer to a p2p network, and there is no real need to bother our own instance unnecessarily.   [actually, if all user's posts were signed and authenticated, I could then in turn make the posts available to my own instance, but that's a discussion for another day].

c. Let us take the limit example - I make my own instance with only me as a user.
There is still a difference between the remote posts my instance knows about (any post by anyone I follow) and the remote local timeline (all local posts even from people I don't follow).  
As a hack, I could make a secondary account on my instance and run a script that continually makes it follow any new account on the remote instance so that their posts also get sent to my instance (and by default shown in federated timeline). This would then allow me to see all posts from the remote instance while still only querying my own instance (except of course for posts of those people that blocked my secondary account)... You see it gets quite convoluted.

**Comment by brendanjones** [comment]
+1 to that @orenTal 's question

@sparr Yes my comment ignores technical limitations - I'm trying to pinpoint what people want to achieve. I welcome any input on why any of them can't be achieved, or can be achieved but require overly large changes. 

Ignoring the technical ramifications of implementing either, answer me this: if you look at the local feed of an instance, would you expect to see it as people from that instance see it, or limited to only the content that's been federated? Surely the former?

**Comment by sparr** [comment]
@brendanjones I want the former, but would accept the latter as an incremental improvement. I also know that the latter would require *far* less code and changes to the architecture and non-technical decisions built into the design of the client.

@orenTal see #21571, find in page "privacy" and "gmail" or just read the whole thing.

**Comment by sparr** [comment]
Elaborating on "far less"... I suspect the former option would require a few dozen lines of code, and most of those would be around prompting and storing the list of instances you want to filter, as a preference or as a new follow type or similar. The actual filtering itself would be a trivial addition to existing filtering functionality.

The latter would require hundreds to thousands of lines of code, as well as changes to high level design decisions from the founders of the project (or implementation in a third party client that doesn't have those concerns).

**Comment by brendanjones** [comment]
@sparr I think you might have flipped your formers and latters between your first and second comment - might need to edit?

**Comment by Terra100f** [comment]
@brendanjones 
> @Terra100f iiuc you feel that lists don’t feel like the place to go for those topics. But when you think about viewing a community or topic, what would you expect to see? A feed of it, right? So to me the issue isn’t that communities and topics aren’t feeds, but rather that you don’t expect to explore those kind of feeds from Lists. And that makes it a naming/education issue. What if lists were changed to be called, say, “Custom feeds”? Would that be the place you’d expect to be able to create and save custom feeds, which could include specific topics? And, to bring it back to this issue, ‘Custom feeds’ would also be where you’d follow other instances?

Sure. I can see a name change being the UX solution.

#4 or #6 above seemed to be possibilities -- it looks like the technical aspects are being chatted about here, which is good as next step.

If "custom feeds" is implemented as replacing the "lists" label, it would likely create subsequent user requests for expanding the expectation of customizing. I think that's good. Users would likely want to customize based on --

- hashtags
- keywords
- **instances**
- people/accounts (current usage)
- trends (which would then have a domino effect of users wanting more from Explore, such as trends beyond the current limited instance scope and not constricted by hashtags) -- I realize this sounds Twitter-ish, but centralization and some algorithm does serve a valuable town square need and helps us get out of siloed thinking

**Comment by Terra100f** [comment]
Whatever solution is implemented, it should probably take into consideration how it would impact the scale problem outlined in this post -- https://sciences.social/@josh/109410524407708551 -- meaning, would the solution create a storage/data/load problem for the user's instance that might upset the sustainability factor?

And, would I be able to like, boost anything in those custom feeds from other instances? (currently cannot)

**Comment by orenTal** [comment]
@sparr I read that thread and the part about Gmail. However, I don't think the cases are equivalent. 
When it comes to email if you put a specific image, then you know a specific reader opens the email. because no one else has the URL of the image. 

So you know that the specific person that you have sent the mail has opened.


In the case of the mastodon client directly fetching feed from other servers, they can know only the IP of the client's computer. This is no different than what they will know if RSS client is used or if one brows directly to the instance website. 

Maybe there is a privacy expert that can expand the issue. but the way I see it, from a privacy perspective, there is no difference if the user manually browsed to every specific instance or use one client to fetch the feed of all the instances. The same information is exposed, and that is only the IP.

**Comment by robsonsobral** [comment]
> [...] the way I see it, from a privacy perspective, there is no difference if the user manually browsed to every specific instance or use one client to fetch the feed of all the instances. The same information is exposed, and that is only the IP.

@orenTal , you answered your own question. There's no difference, and that's the problem. In case the user browse to a website, she choses to do it. Like @brendanjones , I'm going to try to not be too technical.

Let's pretend there's an instance **evil.empire**. You browsed to it, took a look around and realize you don't want them to know nothing about you. Then, while navigating on another instance you find a profile of somebody from the **evil.empire**. The moment you load data from it, the **empire** will know about you. This can be problematic in countries with no respect for individual liberties.

Please, tell me in case I couldn't be clear enough. Devs have a tendency to over explain things.

Given that, I think the approach I suggested on the other issue looks simple enough from a normal human user perspective. (Devs aren't normal people! They're weird!)

1. a button "Load data from **evil.empire**" and a little link "privacy concerns" next to it pointing to the preferences page where it can be set and with an explanation before them;
2. in case the user click the "privacy concerns", show her an option to "Always block direct contact to **evil.empire**" right on the Preferences page;
3. in case the user click the "Load data from", show her an option to "Always trust **evil.empire**.

P.S.: Please, in case I'm repeating something someone already said, forgive me. I can't read all the thread right now. I'm replying because I was indirectly mentioned.

**Comment by Terra100f** [comment]
FYI -- The EU recently created this page. I was able to follow accounts, even from other instances, directly from the page -- https://social.network.europa.eu/explore. It was easy

**Comment by orenTal** [comment]
> FYI -- The EU recently created this page. I was able to follow accounts, even from other instances, directly from the page -- https://social.network.europa.eu/explore. It was easy

It doesn't work for me. when I try to follow, it asks me to sign in or sign up. 

And that is the whole point of the requested feature, that we will be able to see the accounts, and feeds of other instances.

**Comment by cgohier** [comment]
>It doesn't work for me. when I try to follow, it asks me to sign in or sign up.

Interestingly, it works if I'm logged in on mastodon.online.

**Comment by brendanjones** [comment]
Of interest: https://github.com/lartsch/FediFollow-Chrome

**Comment by robsonsobral** [comment]
As this discussion has a broader aspect, I posted my suggestion where there are others developers already [talking about it](https://github.com/mastodon/mastodon/issues/17213#issuecomment-1356870891).

**Comment by ashed** [comment]
Would you implement this functionality: Follow public local timelines from other instances?
I vote in this thread (https://github.com/mastodon/mastodon/issues/6942#issuecomment-1328052594)
Option 5

**Comment by nettbeast** [comment]
> Just what the title says.
> 
> Maybe implemented like the "show a toot from another instance"-feature: Copying the full URL of the instance into the search bar opens the local timeline of this other instance.
> 
> Which makes following interests easier, since a lot of instances are specialised on some specific interests, but a person does'nt follow only one. So to mix things a lot more up.

It would GREATLY encourage organizations to build their own mastodon environments if you could follow the public timeline on any server.   IE maybe the Washington Post creates "mastodon.washingtonpost.com" and I could subscribe and follow the whole thing if I wanted to...

**Comment by cgohier** [comment]
> t would GREATLY encourage organizations to build their own mastodon environments if you could follow the public timeline on any server. IE maybe the Washington Post creates "mastodon.washingtonpost.com" and I could subscribe and follow the whole thing if I wanted to...

It is already possible in some clients. FediLab for example.

**Comment by jadedgnome** [comment]
yeah on reddit it's possible to combine two or more subreddits so users can view submissions for both simultaneously.

https://www.reddit.com/r/privacy + https://www.reddit.com/r/PrivacyGuides = **https://www.reddit.com/r/privacy+PrivacyGuides**

it would be nifty to be able to do something similar on mastodon.

I'd want to combine Local feeds from niche but similar instances so I can scroll through 2+ local feeds simultaneously.

**Comment by sami-huvilanet** [comment]
Following steadily other Local timelines should be possible  **if an instance would allow it** and not block followings from elsewhere.

Some selective special instances would likely –and understandably– be more protective with their Local feeds.

**Comment by KaKi87** [comment]
There's already access control for `/public/local` as well as equivalent API endpoints.

As long as the route is available, then it's allowed for use.

This feature would only bring the convenience of not requiring the user to copy/paste profile/post URLs to interact with those.

**Comment by KaKi87** [comment]
So, since this doesn't seem to be happening anytime soon, I'm creating [an open source webapp](https://git.kaki87.net/KaKi87/mastovuepp) that will allow this :

![](https://user-images.githubusercontent.com/21284089/236621686-b4851641-fc27-4cf7-b3c0-3fe6f5a9a1b9.png)

It's not ready yet (unusable on mobile, unimplemented menu) but I'll release it at [mastovuepp.kaki87.net](https://mastovuepp.kaki87.net) as soon as it is.

**Comment by Pxtl** [comment]
In my above-linked feature I have a specific idea for how the UI for this could work:

1) Make `#@Instancename` a special hashtag that's implicitly automatically applied to every post on an instance.
2) Make the "local" feed a search for `#@Instancename`
3) Allow admins/moderators of `@instancename` to moderate `#@instancename` to control posts on there that come from other instances.

Then users can manually search, post, and follow `#@Instancename` as a hashtag, with no additional front-end UI needed.

**Comment by LuNeder** [comment]
This would be similar to Akkoma’s Bubble Timeline feature, which is probably a good inspiration for how to implement this

**Comment by KaKi87** [comment]
I don't know that one, I also discovered the [*antenna* feature from Firefish](https://firefish.social/notes/9b5uy76taw) a while ago.

**Comment by LuNeder** [comment]
> I don’t know that one

It’s quite similar to what this issue asks for tbh. It’s like the local timeline, but from more than one instance (handpicked by the instance admin tho). It shows toots from the local timeline of the selected instances.
I guess the only difference is that on this case I think the bubble timeline participants is selected by the instance admins.

**Comment by KaKi87** [comment]
> handpicked by the instance admin tho

Oh, that's not what's wanted here then.

Meanwhile, Firefish's antenna feature does allow users to select which instances they want to see posts from.

Thanks

**Comment by omentic** [comment]
I think this would pair particularly nicely with the exclusive lists that were recently added - one could follow a timeline, add it to an exclusive list (so as to remove it from the default home feed), and then switch to it in the sidebar whenever.

**Comment by eddeemn** [comment]
I'm amazed that this has been an open discussion for six years. Is there any progress on this or possibility of it happening?
